magefiles/rulesengine/repos/infra_deployments.go

```diff
@@ -1,17 +1,166 @@
+
+		return len(rctx.DiffFiles.FilterByDirGlob("components/image-controller/**/*")) != 0
+	}),
+	Actions: []rulesengine.Action{rulesengine.ActionFunc(func(rctx *rulesengine.RuleCtx) error {
+		AddLabelToLabelFilter(rctx, "image-controller")
+		return nil
+	})}}
+
+var InfraDeploymentsMultiPlatformComponentChangeRule = rulesengine.Rule{Name: "Infra-deployments PR Multi Controller component File Change Rule",
+	Description: "Map multi platform tests files when Multi Controller component files are changed in the infra-deployments PR",
+	Condition: rulesengine.ConditionFunc(func(rctx *rulesengine.RuleCtx) bool {
+
+		return len(rctx.DiffFiles.FilterByDirGlob("components/multi-platform-controller/**/*")) != 0
+	}),
+	Actions: []rulesengine.Action{rulesengine.ActionFunc(func(rctx *rulesengine.RuleCtx) error {
+		AddLabelToLabelFilter(rctx, "multi-platform")
+		return nil
+	})}}
+
+var InfraDeploymentsBuildTemplatesComponentChangeRule = rulesengine.Rule{Name: "Infra-deployments PR Build-templates component File Change Rule",
+	Description: "Map build-templates tests files when Build-templates component files are changed in the infra-deployments PR",
+	Condition: rulesengine.ConditionFunc(func(rctx *rulesengine.RuleCtx) bool {
+
+		return len(rctx.DiffFiles.FilterByDirGlob("components/build-service/base/build-pipeline-config/*")) != 0
+	}),
+	Actions: []rulesengine.Action{rulesengine.ActionFunc(func(rctx *rulesengine.RuleCtx) error {
+		AddLabelToLabelFilter(rctx, "build-templates")
+		return nil
+	})}}
+
+var InfraDeploymentsBuildServiceComponentChangeRule = rulesengine.Rule{Name: "Infra-deployments PR Build service component File Change Rule",
+	Description: "Map build service tests files when Build service component files are changed in the infra-deployments PR",
+	Condition: rulesengine.ConditionFunc(func(rctx *rulesengine.RuleCtx) bool {
+
+		return len(rctx.DiffFiles.FilterByDirGlob("components/build-service/**/*")) != 0 &&
+			len(rctx.DiffFiles.FilterByDirGlob("components/build-service/base/build-pipeline-config/*")) == 0
+	}),
+	Actions: []rulesengine.Action{rulesengine.ActionFunc(func(rctx *rulesengine.RuleCtx) error {
+		AddLabelToLabelFilter(rctx, "build-service")
+		return nil
+	})}}
+
+var InfraDeploymentsBuildServiceTemplatesComponentChangeRule = rulesengine.Rule{Name: "Infra-deployments PR Build service component and templates File Change Rule",
+	Description: "Map build service tests files when Build service component and templates files are changed in the infra-deployments PR",
+	Condition: rulesengine.ConditionFunc(func(rctx *rulesengine.RuleCtx) bool {
+
+		return len(rctx.DiffFiles.FilterByDirGlob("components/build-service/**/*")) != 0 &&
+			len(rctx.DiffFiles.FilterByDirGlob("components/build-service/base/build-pipeline-config/*")) != 0
+	}),
+	Actions: []rulesengine.Action{rulesengine.ActionFunc(func(rctx *rulesengine.RuleCtx) error {
+		AddLabelToLabelFilter(rctx, "build-service")
+		return nil
+	})}}
```

Comment:
I think this is unnecessary rule and will lead to applying `build-service` rule even when only files under `components/build-service/base/build-pipeline-config/*` are changed.

But IMO this is a behavior we can live with for a while until we fix this.

Reply:
thanks @psturc , 
I've tested it locally and you are right , 
please find the fix https://github.com/konflux-ci/e2e-tests/pull/1347